src/tof/chopper.py

```diff
@@ -54,26 +56,40 @@ def omega(self) -> sc.Variable:
         """
         The angular velocity of the chopper.
         """
-        return sc.constants.pi * (2.0 * sc.units.rad) * self.frequency
+        return two_pi * self.frequency
 
-    @property
-    def open_times(self) -> sc.Variable:
-        """
-        The times at which the chopper is open.
+    def open_close_times(
+        self, time_limit: sc.Variable
+    ) -> Tuple[sc.Variable, sc.Variable]:
         """
-        return (
-            self.open.to(unit='rad', copy=False) + self.phase.to(unit='rad', copy=False)
-        ) / self.omega
+        The times at which the chopper opens and closes.
 
-    @property
-    def close_times(self) -> sc.Variable:
-        """
-        The times at which the chopper is closed.
+        Parameters
+        ----------
+        time_limit:
+            Determines how many rotations the chopper needs to perform to reach the time
+            limit.
         """
+        open_times = []
+        close_times = []
+        nrot = 0
+        phase = self.phase.to(unit='rad')
+        time_limit = time_limit.to(unit='s')
+        while True:
+            nrot += 1
+            open_times.append(
+                (self.open.to(unit='rad', copy=False) + phase) / self.omega
+            )
+            close_times.append(
+                (self.close.to(unit='rad', copy=False) + phase) / self.omega
+            )
+            if close_times[-1].max() > time_limit:
+                break
+            phase += two_pi
         return (
-            self.close.to(unit='rad', copy=False)
-            + self.phase.to(unit='rad', copy=False)
-        ) / self.omega
+            sc.concat(open_times, self.open.dim),
+            sc.concat(close_times, self.close.dim),
+        )
```

Comment:
Not nice... have you considered making a `Variable` from phase increments, and let scipp broadcasting do the rest? Might overshoot by only doing full chopper rotations, but seems simpler?